### PR TITLE
apptarget not available in webworker

### PR DESCRIPTION
**Pull request by pelikhan**

### CODE REVIEW

File: pxtcompiler/emitter/hexfile.ts [hunk 1]
```diff
@@ -132,10 +132,10 @@ namespace ts.pxtc {
             }
         }
 
-        export function encodeVTPtr(ptr: number) {
-            let vv = ptr >> pxt.appTarget.compile.vtableShift
+        export function encodeVTPtr(ptr: number, opts: CompileOptions) {
```
Comment (riknoll) [inline pxtcompiler/emitter/hexfile.ts]: Is this function never used?

Comment (pelikhan) [reply pxtcompiler/emitter/hexfile.ts]: Maybe used in another branch by Michal.

Comment (riknoll) [reply pxtcompiler/emitter/hexfile.ts]: Looks like it's used in emitter/backvm.ts

**Comment by mmoskal** [comment]
Thanks!

BTW, why don't we just make the `appTarget` available in the worker?

**Comment by pelikhan** [comment]
We do modify apptarget in the editor so that would not work anymore.

**Pull request merged**